commands/audit/scarunner.go

```diff
@@ -109,9 +98,25 @@ func buildDepTreeAndRunScaScan(auditParallelRunner *utils.SecurityParallelRunner
 			_ = targetResult.AddTargetError(fmt.Errorf("failed to build dependency tree: %s", bdtErr.Error()), auditParams.AllowPartialResults())
 			continue
 		}
+		if auditParams.diffMode {
+			var ddtErr error
+			if auditParams.resultsToCompare != nil {
+				treeResult, ddtErr = getDiffDependencyTree(targetResult, auditParams.resultsToCompare, treeResult)
+				if ddtErr != nil {
+					_ = targetResult.AddTargetError(fmt.Errorf("failed to build diff dependency tree: %s", ddtErr.Error()), auditParams.AllowPartialResults())
+					continue
+				}
+			} else {
+				log.Debug(fmt.Sprintf("Diff scan - calculated dependencies tree for target %s in source branch, skipping scan part",
+					targetResult.Target))
+				continue
+			}
+		}
+		if err := logDeps(treeResult.FlatTree); err != nil {
+			log.Warn("Failed to log dependencies tree: " + err.Error())
+		}
```

Comment:
```suggestion
		if auditParams.diffMode {
			if auditParams.resultsToCompare != nil {
				if treeResult, actionErr = getDiffDependencyTree(targetResult, auditParams.resultsToCompare, treeResult); actionErr != nil {
					_ = targetResult.AddTargetError(fmt.Errorf("failed to build diff dependency tree: %s", actionErr.Error()), auditParams.AllowPartialResults())
					continue
				}
			} else {
				log.Debug(fmt.Sprintf("Diff scan - calculated dependencies tree for target %s in source branch, skipping scan part",
					targetResult.Target))
				continue
			}
		}
		if err := logDeps(treeResult.FlatTree); err != nil {
			log.Warn("Failed to log dependencies tree: " + err.Error())
		}
```

I would use `bdtErr` (maybe refactor name) and not a new var, it is not needed anymore and will make the code more readable
maybe I would extract this to a func so we can reduce nested if's